code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -245,8 +245,20 @@ static final class NotificationVisibility {
                 DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.CHANNEL_ID, null);
         this.actionUri =
                 DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.ACTION_URI, null);
-        this.smallIcon =
+        final String smallIcon =
                 DataReader.optString(data, CampaignPushConstants.PushPayloadKeys.SMALL_ICON, null);
+        if (StringUtils.isNullOrEmpty(smallIcon)) {
+            Log.debug(
+                    CampaignPushConstants.LOG_TAG,
+                    SELF_TAG,
+                    "The \"adb_small_icon\" key is not present in the message data payload, using"
+                            + " the \"adb_icon\" key instead.");
+            this.smallIcon =
+                    DataReader.optString(
+                            data, CampaignPushConstants.PushPayloadKeys.LEGACY_SMALL_ICON, null);
+        } else {
+            this.smallIcon = smallIcon;
+        }
```

Comment:
nit: I don't think this else part is required. 
else is executed when `smallIcon` is nil or empty. And by default this.smallIcon is null. We can just leave it that way



May be let's tweak the log comment a little bit (because we are just attempting to use legacy icon. It can still be null )
`The \"adb_small_icon\" key is not present in the message data payload, using the \"adb_icon\" key instead.`

to

`The \"adb_small_icon\" key is not present in the message data payload, attempting to use \"adb_icon\" key instead.`